contracts/voting-verifier/src/execute.rs

```diff
@@ -167,18 +167,29 @@ pub fn vote(
     poll_id: PollID,
     votes: Vec<bool>,
 ) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
     let mut poll = POLLS
         .may_load(deps.storage, poll_id)?
         .ok_or(ContractError::PollNotFound)?;
     match &mut poll {
         state::Poll::Messages(poll) | state::Poll::ConfirmWorkerSet(poll) => {
-            poll.cast_vote(env.block.height, &info.sender, votes)?
+            poll.cast_vote(env.block.height, config.grace_period, &info.sender, votes)?
         }
     };
 
     POLLS.save(deps.storage, poll_id, &poll)?;
 
-    Ok(Response::new().add_event(
+    let rewards_msg = WasmMsg::Execute {
+        contract_addr: config.rewards_contract.into_string(),
+        msg: to_binary(&rewards::msg::ExecuteMsg::RecordParticipation {
+            event_id: poll_id.into(),
+            worker_address: info.sender.to_string(),
+        })?,
+        funds: vec![],
+    };
+
+    Ok(Response::new().add_message(rewards_msg).add_event(
```

Comment:
We shouldn't do this blindly here. Workers are only supposed to be rewarded if they vote according to consensus. So if consensus is true, we only reward workers who voted true.

Reply:
yeah I think this needs to be moved to the poll completion call, so we know who to reward